### PR TITLE
fix running non-interactive commands on agentless nodes

### LINKED ISSUE #23941: `tsh ssh command` fails for agentless nodes
Expected behavior:

`tsh ssh node ls` works.

Current behavior:

```bash
% ssh ubuntu@node ls
... ls output ...
% tsh ssh ubuntu@node ls
Failed to launch: user: unknown user ubuntu.
ERROR: Process exited with status 255
```

Bug details:
- Teleport version 13.0-dev
- Recreation steps
- Debug logs

### PR DESCRIPTION
**Pull request by capnspacehook**
Fixes https://github.com/gravitational/teleport/issues/23941.

### CODE REVIEW

File: integration/integration_test.go [hunk 1]
```diff
@@ -7338,17 +7339,26 @@ func testAgentlessConnection(t *testing.T, suite *integrationTestSuite) {
...
+		_ = nodeClient.Close()
+	})
 
 	// forward SSH agent
 	sshClient := nodeClient.Client.Client
 	session, err := sshClient.NewSession()
 	require.NoError(t, err)
 	t.Cleanup(func() {
-		require.NoError(t, session.Close())
+		// the error is ignored here to avoid failing on io.EOF
```
Comment (jakule) [inline integration/integration_test.go:7352]: Why did we move from having no error to returning EOF sometimes?

**Review by r0mant — APPROVED** [review]

File: integration/integration_test.go [hunk 2]
```diff
@@ -7409,11 +7420,19 @@ func startSSHServer(t *testing.T, caPubKeys []ssh.PublicKey, hostKey ssh.Signer)
 				}
 				if req.Type == sshutils.AgentForwardRequest {
 					agentForwarded = true
+				} else if req.Type == sshutils.ExecRequest {
+					_, err = channel.SendRequest("exit-status", false, binary.BigEndian.AppendUint32(nil, 0))
```
Comment (jakule) [inline integration/integration_test.go:7424]: Why are we always sending exit-code 0? I don't mind hardcoding a value in a test itself, but this is a test utility function used in many places. Hardcoding a value here maybe really confusing for someone who will use this function the next time.

File: lib/srv/exec.go [hunk 3]
```diff
@@ -93,9 +94,9 @@ func NewExecRequest(ctx *ServerContext, command string) (Exec, error) {
 		}, nil
 	}
 
-	// When in recording mode, return an *remoteExec which will execute the
-	// command on a remote host. This is used by in-memory forwarding nodes.
-	if services.IsRecordAtProxy(ctx.SessionRecordingConfig.GetMode()) {
+	// If this is not a Teleport node, execute the command on a remote
+	// host. This is used by in-memory forwarding nodes.
+	if ctx.ServerSubKind == types.SubKindOpenSSHNode || services.IsRecordAtProxy(ctx.SessionRecordingConfig.GetMode()) {
```
Comment (jakule) [inline lib/srv/exec.go:99]: Comment and condition are different. `Teleport Node == (!RecordingMode && !OpenSSH)` only until a new type is added.

**Review by codingllama — APPROVED** [review]

File: integration/integration_test.go (hunk 1 above)
Comment (capnspacehook) [reply integration/integration_test.go:7352]: Because the ssh server from `startSSHServer` now closes the session to avoid a deadlock, I'll comment that

Comment (capnspacehook) [reply integration/integration_test.go]: Done

File: lib/srv/exec.go (hunk 3 above)
Comment (capnspacehook) [reply lib/srv/exec.go]: Good catch, done

**Review by jakule — APPROVED** [review]

**Pull request merged**